src/main/scala/edu/ie3/simona/event/ResultEvent.scala

```diff
@@ -131,6 +132,26 @@ object ResultEvent {
     }
   }
 
+  object DomesticHotWaterStorageResult {
+    def unapply(result: DomesticHotWaterStorageResult): Option[
+      (
+          ZonedDateTime,
+          UUID,
+          ComparableQuantity[Power],
+          ComparableQuantity[Energy],
+      )
+    ] = {
+      Option(result).map { result =>
+        (
+          result.getTime,
+          result.getInputModel,
+          result.getqDot,
+          result.getEnergy,
+        )
+      }
+    }
+  }
```

Comment:
Maybe this could be combined with `CylindricalThermalStorageResult`, since both extend `AbstractThermalStorageResult` and have the same information.